piq/mdsi.py

```diff
@@ -0,0 +1,184 @@
+    prediction, target = _adjust_dimensions(input_tensors=(prediction, target))
+
+    if prediction.size(1) == 1:
+        prediction = prediction.repeat(1, 3, 1, 1)
+        target = target.repeat(1, 3, 1, 1)
+        warnings.warn('The original VSI supports only RGB images. The input images were converted to RGB by copying '
+                      'the grey channel 3 times.')
+
+    prediction = prediction * 255. / data_range
+    target = target * 255. / data_range
+
+    # Averaging image if the size is large enough
+    kernel_size = max(1, round(min(prediction.size()[-2:]) / 256))
+    padding = kernel_size // 2
+
+    if padding:
+        up_pad = (kernel_size - 1) // 2
+        down_pad = padding
+        pad_to_use = [up_pad, down_pad, up_pad, down_pad]
+        prediction = pad(prediction, pad=pad_to_use)
+        target = pad(target, pad=pad_to_use)
+
+    prediction = avg_pool2d(prediction, kernel_size=kernel_size)
+    target = avg_pool2d(target, kernel_size=kernel_size)
+
+    prediction_lhm = rgb2lhm(prediction)
+    target_lhm = rgb2lhm(target)
+
+    kernels = torch.stack([prewitt_filter(), prewitt_filter().transpose(1, 2)]).to(prediction)
+    gm_prediction = gradient_map(prediction_lhm[:, :1], kernels)
+    gm_target = gradient_map(target_lhm[:, :1], kernels)
+    gm_avg = gradient_map((prediction_lhm[:, :1] + target_lhm[:, :1]) / 2., kernels)
+
+    gs_prediction_target = similarity_map(gm_prediction, gm_target, c1)
+    gs_prediction_average = similarity_map(gm_prediction, gm_avg, c2)
+    gs_target_average = similarity_map(gm_target, gm_avg, c2)
+
+    gs_total = gs_prediction_target + gs_prediction_average - gs_target_average
+
+    cs_total = (2 * (prediction_lhm[:, 1:2] * target_lhm[:, 1:2] +
+                     prediction_lhm[:, 2:] * target_lhm[:, 2:]) + c3) / (prediction_lhm[:, 1:2] ** 2 +
+                                                                         target_lhm[:, 1:2] ** 2 +
+                                                                         prediction_lhm[:, 2:] ** 2 +
+                                                                         target_lhm[:, 2:] ** 2 + c3)
+
+    if combination == 'sum':
+        gcs = (alpha * gs_total + (1 - alpha) * cs_total)
+    elif combination == 'mult':
+        gs_total_pow = pow_for_complex(base=gs_total, exp=gamma)
+        cs_total_pow = pow_for_complex(base=cs_total, exp=beta)
+        gcs = torch.stack((gs_total_pow[0] * cs_total_pow[0], gs_total_pow[1] + cs_total_pow[1]))
+    else:
+        raise ValueError(f'Expected combination method "sum" or "mult", got {combination}')
+
+    mct_complex = pow_for_complex(base=gcs, exp=q).mean(dim=(-1, -2), keepdim=True)
+    score = (pow_for_complex(base=gcs, exp=q) - mct_complex).pow(2).sum(dim=0).sqrt()
+    score = (score ** rho).mean(dim=(-1, -2)) ** (o / rho)
+    if reduction == 'none':
+        return score
+    return {'mean': score.mean,
+            'sum': score.sum}[reduction](dim=0)
```

Comment:
Please check return tensor shape. All other metrics return tensor of size 0, while this metric returns tensor with size 1. 
Minimal example: 
```python
import piq
import torch
prediction = torch.rand(4, 3, 256, 256)
target = torch.rand(4, 3, 256, 256)

piq.mdsi(prediction, target).shape, piq.ssim(prediction, target).shape
```

Result: `(torch.Size([1]), torch.Size([]))`